src/components/with-suggestions/redux/reducer.ts

```diff
@@ -0,0 +1,48 @@
+import { fromJS } from 'immutable';
+
+import * as actions from './actions';
+import {
+  GET_SEARCH_SUGGESTIONS_REQUESTED,
+  GET_SEARCH_SUGGESTIONS_SUCCEEDED,
+  GET_SEARCH_SUGGESTIONS_FAILED,
+  RESET_SEARCH_SUGGESTIONS
+} from './action-types';
+
+import type { Actions, SearchSuggestion } from '../../../types';
+
+const initialState = fromJS({
+  searchSuggestions: null,
+  isLoadingSuggestions: false,
+  cachedSuggestions: {} as Record<string, SearchSuggestion[]>
+});
+
+export default function reducer(
+  state: any = initialState,
+  action: Actions<typeof actions>
+) {
+  switch (action.type) {
+    case GET_SEARCH_SUGGESTIONS_REQUESTED:
+      return state
+        .set('searchSuggestions', null)
+        .set('isLoadingSuggestions', true);
+    case GET_SEARCH_SUGGESTIONS_SUCCEEDED:
+      return state
+        .set(
+          'cachedSuggestions',
+          fromJS({
+            ...state.get('cachedSuggestions').toJS(),
+            [action.payload.query]: action.payload.suggestions
+          })
+        )
+        .set('searchSuggestions', fromJS(action.payload.suggestions))
```

Comment:
Jeg ser ikke at det er nødvendig å lagre suggestions to ganger, det vil vel holde å bare ha "cachedSuggestions" ?